crates/oxc_cli/src/result.rs

```diff
@@ -7,6 +7,7 @@ use std::{
 pub enum CliRunResult {
     None,
     PathNotFound { paths: Vec<PathBuf> },
+    MaxWarningsExceeded { number_of_warnings: usize, max_warnings: usize },
     LintResult { number_of_files: usize, number_of_diagnostics: usize },
```

Comment:
hmmm, I think these belong together, we still want to print the total number of files.

We can change `number_of_diagnostics` to `number_of_warnings` here.